src/core/rag/confidence/PerformanceOptimizer.ts

```
@@ -156,11 +172,14 @@ export class PerformanceOptimizer {
       if (!this.batchQueue.has(batchKey)) {
         this.batchQueue.set(batchKey, []);
       }
-      this.batchQueue.get(batchKey)!.push(request);
+      const queue = this.batchQueue.get(batchKey);
+      if (queue) {
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)